src/vscode-dts/vscode.proposed.dataTransferFiles.d.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/TODO
+
+	interface DataTransferFile {
+		/**
+		 * The name of the file.
+		 */
+		readonly name: string;
+
+		/**
+		 * The full file path of the file.
+		 *
+		 * May be undefined on web.
+		 */
+		readonly uri?: Uri;
+
+		/**
+		 * The full file contents of the file.
+		 */
+		data(): Thenable<Uint8Array>;
```

Comment:
The [dom API](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/getAsFile) returns a `File` with `getAsFile()`, so this makes sense to me. However, the dom API offers both a [`stream()` and `text()` methods](https://developer.mozilla.org/en-US/docs/Web/API/File#instance_methods) on `File`. Should we also have a `stream()` method to encourage extension authors not to read whole files into memory? Since the file will need to be transered cross process (or across the network in the case of remote), supporting files at all could be quite expensive and slow.

Reply:
I've pushed a change to make the file transfer lazy

However I did not yet  add a way to read part of the file contents. Our existing fs apis also only allow working with whole files, so I wanted to stay consistent with them